src/Pages/CreateNewCharacterPage/CreateNewCharacterPage.test.tsx

```diff
@@ -0,0 +1,184 @@
+});
+
+test('if clicking on the reset character button resets all the values on the page', async () => {
+  const { user } = setUpTest();
+
+  const searchbar = screen.getByRole('combobox');
+  await selectEvent.select(searchbar, ['Absolute Direction']);
+
+  await user.click(screen.getByRole('button', { name: 'Reset Character' }));
+  expect(screen.queryAllByText('Absolute Direction').length).toBe(0);
+  expect(
+    screen.getByRole('heading', { name: 'Select your Characters ADVANTAGES' })
+  ).toBeInTheDocument();
+});
+
+test('if clicking on the save character button calls the createCharacterDocument function with correct arguments', async () => {
+  mockGetUsersCharactersList.mockResolvedValue(['not duplicate']);
+  mockAddNewCharacter.mockResolvedValue(mockCharacterDocument);
+  const { user } = renderWithProviders(<CreateNewCharacterPage />, preloadedState);
+
+  const saveCharacterButton = screen.getByRole('button', { name: 'Save Character' });
+  await user.click(saveCharacterButton);
+  expect(mockCreateCharacterDoc).toHaveBeenCalledWith(mockCharacterDocument, 'testid');
+});
+
+test('if successful save alert is displayed only after the character is saved', async () => {
+  mockGetUsersCharactersList.mockResolvedValue(['not duplicate']);
+  mockAddNewCharacter.mockResolvedValue(mockCharacterDocument);
+  const { user } = renderWithProviders(<CreateNewCharacterPage />, preloadedState);
+
+  expect(screen.queryByRole('alert')).not.toBeInTheDocument();
+
+  const saveCharacterButton = screen.getByRole('button', { name: 'Save Character' });
+  await user.click(saveCharacterButton);
+  expect(await screen.findByRole('alert')).toHaveTextContent(
+    'Test Character has been succesfully saved.'
+  );
+});
+
+describe('if the correct alerts are displayed when a save character requirement is not met', () => {
+  test('duplicate character name', async () => {
+    mockRetrieveCharacterList.mockResolvedValue(['Test Character']);
+    const { user } = renderWithProviders(<CreateNewCharacterPage />, preloadedState);
+
+    await user.click(screen.getByRole('button', { name: 'Save Character' }));
+  });
+
+  test('no character name entered', async () => {
+    mockRetrieveCharacterList.mockResolvedValue(['Test Character']);
+    const { user } = setUpTest();
+
+    await user.click(screen.getByRole('button', { name: 'Save Character' }));
+    expect(screen.getByRole('alert')).toHaveTextContent(
+      'You must type a name for your character and press enter in order to save'
+    );
+  });
+
+  test('no attributes selected', async () => {
+    mockRetrieveCharacterList.mockResolvedValue(['Test Character']);
+    const { user } = setUpTest();
+
+    const nameTextInput = screen.getByRole('textbox', { name: 'character-name-form' });
+    await user.clear(nameTextInput);
+    await user.type(nameTextInput, 'test name');
+    await user.keyboard('{Enter}');
+
+    await user.click(screen.getByRole('button', { name: 'Save Character' }));
```

Comment:
Tests are looking really good man, good work. sometimes you've just gotta deal with some duplication, sure if you can see a bunch try and abstract it out. you might end up with quite a few util like functions in the test file or shared across you could put in a util.ts file. or you you just duplicate heaps, kinda how most go

Reply:
After seeing more test examples it looks like duplication is the prefered way to go, I have gotten over my urge to abstract more and just go with it now.
I think it does make figuring ou what is being tested in the end.